Sources/SignalProducer.swift

```diff
@@ -65,6 +65,23 @@ public struct SignalProducer<Value, Error: Swift.Error> {
 		}
 	}
 
+	/// Creates a producer for a `Signal` that immediately sends one value, then
+	/// completes.
+	///
+	/// This initializer differs from `init(value:)` in that it will not 
+	/// evaluate the closure supplying its sole `value` event until the 
+	/// `SignalProducer` is started.
+	///
+	/// - parameters:
+	///   - block: A block that supplies a value to be sent by the `Signal` in
+	///            a `value` event.
+	public init(_ block: @escaping () -> Value) {
```

Comment:
@NachoSoto In a nutshell, `@autoclosure` implies `@noescape`, and hence we lose out on the ability to defer execution.